### PR TITLE
Ra 37 external alias

### PR DESCRIPTION
**Pull request by ingeknudsen**
Will make it possible to add radix config like this:

  dnsExternalAlias:
    - alias: test1.iknu.stian.tech
      component: frontend
      environment: dev
    - alias: test2.iknu.stian.tech
      component: frontend
      environment: dev

### CODE REVIEW

File: pkg/apis/deployment/secrets.go [hunk 1]
```diff
@@ -84,6 +167,18 @@ func (deploy *Deployment) garbageCollectSecretsNoLongerInSpec() error {
 	return nil
 }
 
+func (deploy *Deployment) listSecretsForComponent(component radixv1.RadixDeployComponent, labelSelector string) (*v1.SecretList, error) {
```
Comment (thezultimate) [inline pkg/apis/deployment/secrets.go]: `component` variable is not used in this function. Remove it?

File: pkg/apis/radixvalidators/validate_ra.go [hunk 2]
```diff
@@ -83,6 +89,41 @@ func validateDNSAppAlias(app *radixv1.RadixApplication) []error {
 	return errs
 }
 
+func validateDNSExternalAlias(app *radixv1.RadixApplication) []error {
+	errs := []error{}
+
+	distictAlias := make(map[string]bool)
```
Comment (thezultimate) [inline pkg/apis/radixvalidators/validate_ra.go]: Possible typo: `distinctAlias`

File: pkg/apis/deployment/secrets.go (hunk 1 above)
Comment (ingeknudsen) [reply pkg/apis/deployment/secrets.go]: Refactored as suggested

File: pkg/apis/radixvalidators/validate_ra.go (hunk 2 above)
Comment (ingeknudsen) [reply pkg/apis/radixvalidators/validate_ra.go]: Renamed, as suggested

**Pull request merged**